libraries/DataFlash/DataFlash_MAVLink.cpp

```diff
@@ -121,7 +127,13 @@ bool DataFlash_MAVLink::WritePrioritisedBlock(const void *pBuffer, uint16_t size
         return false;
     }
 
+    if (!semaphore->take_nonblocking()) {
```

Comment:
Do we really want to use non-blocking take? Would it be so bad to use it with a 10ms timeout?